src/backend/joanie/payment/backends/lyra/__init__.py

```diff
@@ -87,12 +87,12 @@ def _get_common_payload_data(self, order, billing_address=None, installment=None
 
         if billing_address:
             payload["customer"]["billingDetails"] = {
-                "firstName": billing_address["first_name"],
-                "lastName": billing_address["last_name"],
-                "address": billing_address["address"],
-                "zipCode": billing_address["postcode"],
-                "city": billing_address["city"],
-                "country": billing_address["country"],
+                "firstName": billing_address.first_name,
+                "lastName": billing_address.last_name,
+                "address": billing_address.address,
+                "zipCode": billing_address.postcode,
+                "city": billing_address.city,
+                "country": billing_address.country.code,
                 "language": order.owner.language,
             }
```

Comment:
There is a deeper issue here...

The endpoint `submit_installment_payment` is currently passing an address as an object.
The endpoint `submit` is currently passing an address as a dict.

Our modification is currently fixing the `submit_installment_payment` but breaking the `submit` endpoint... because `_get_common_payload` is called by `tokenize`, `create_payment`, `create_one_click_payment` methods.

`tokenize` is used by the endpoint `submit_installment_payment`
`create_payment` and `create_one_click_payment` are used by the endpoint `submit`

So to the resume, when submitting an order, currently, the `_get_common_payload` is called with a billing address which is a dict. When paying a installment, the `_get_common_payload` is called with a billing address which is an object...

Reply:
Okay so i assume we should pass a dictionary of billing address from where we call the method : `create_payment` or `create_one_click_payment` within the viewsets.

Reply:
On the submit method, you could do something like that : 
https://github.com/openfun/joanie/blob/f483785884c581e087f4d1e895c0f508b431133c/src/backend/joanie/core/api/client/__init__.py#L415

```py
        billing_address = models.Address(request.data.get("billing_address"))
```